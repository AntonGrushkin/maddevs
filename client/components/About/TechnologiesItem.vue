<template>
  <div class="technologies-item">
    <h4 class="technologies-item__title sub-title">{{ title }}</h4>
    <p class="technologies-item__paragraph paragraph">{{ paragraph }}</p>
    <div class="icons-list">
      <div class="icon-flex_group">
        <TechIcon v-for="(icon, i) in icons" :key="i" :class-prefix="icon.prefix" :title="icon.title"/>
      </div>
    </div>
  </div>
</template>

<script>


import TechIcon from '../IconsGroup/tech-icon';
export default {
  name: 'TechnologiesItem',
  components: {TechIcon},
  props: {
    title: {
      type: String,
      default: 'Title'
    },
    paragraph: {
      type: String,
      default: 'Paragraph'
    },
    icons: {
      type: Array
    }
  }
};
</script>

<style lang="scss" scoped>
  @import '../../assets/styles/vars';

  .technologies-item {
    padding-bottom: 3vw;

    &__title,
    &__paragraph {
      color: $text-color--grey;
    }

    /deep/.icon-flex_group {
      display: flex;
      flex-wrap: wrap;
    }

    /deep/.icon-item {
      padding-bottom: 15px;
      padding-right: 15px;
    }

    /deep/.icons-list {
      padding: 1.1vw 0;
    }
  }

  .backend {
    padding-bottom: 13px;
  }

  @media only screen and (max-width: 1440px) {
    .technologies-project-management {
      .paragraph {
        padding-right: 65px;
      }
    }
  }

  @media only screen and (max-width: 1280px) {
    .technologies-project-management {
      .paragraph {
        padding-right: 5px;
      }
    }

    .technologies-backend {
      /deep/.icon-flex_group {
        flex-direction: column;
        flex-wrap: initial;
      }
    }
  }

  @media only screen and (max-width: 940px) {
    .technologies-item,
    .technologies-backend {
      /deep/.icon-flex_group {
        flex-direction: row;
      }
    }
  }

  @media only screen and (max-width: 745px) {

  }

  @media only screen and (max-width: 640px) {
    .technologies-backend {
      /deep/.icon-flex_group {
        flex-direction: row;
        flex-wrap: wrap;
      }
    }

    .technologies-item {
      /deep/.icons-list {
        padding: 3.1vw 0;
      }
    }
  }

  @media only screen and (max-width: 520px) {
    .technologies-item {
      padding-bottom: 25px;
    }
  }
</style>
