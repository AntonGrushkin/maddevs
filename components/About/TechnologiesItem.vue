<template>
  <div>
    <div class="technologiesItem">
      <h4 class="technologiesItem__title sub-title">{{ title }}</h4>
      <p class="technologiesItem__paragraph paragraph">{{ paragraph }}</p>
      <BackendIcons v-if="title === 'Backend'"/>
      <FrontendIcons v-if="title === 'Frontend'"/>
      <MobileIcons v-if="title === 'Mobile'"/>
      <ProjectManagement v-if="title === 'Project management'"/>
    </div>
  </div>
</template>

<script>
import BackendIcons from '@/components/IconsGroup/BackendIcons';
import FrontendIcons from '@/components/IconsGroup/FrontendIcons';
import MobileIcons from '@/components/IconsGroup/MobileIcons';
import ProjectManagement from '@/components/IconsGroup/ProjectManagement';

export default {
  name: 'TechnologiesItem',
  props: {
    title: {
      type: String,
      default: 'Title'
    },
    paragraph: {
      type: String,
      default: 'Paragraph'
    }
  },
  components: {
    BackendIcons,
    FrontendIcons,
    MobileIcons,
    ProjectManagement
  }
};
</script>

<style lang="scss" scoped>
  @import '../../assets/styles/vars';
</style>
