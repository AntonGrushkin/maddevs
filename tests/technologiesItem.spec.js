import { mount } from '@vue/test-utils';
import TechnologiesItem from '@/components/About/TechnologiesItem.vue';

describe('TechnologiesItem component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(TechnologiesItem, {
      propsData: {
        title: 'Title',
        paragraph: 'Paragraph'
      }
    });
  });

  test('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('correctly passed props data', () => {
    expect(wrapper.props().title).toBe('Title');
    expect(wrapper.props().paragraph).toBe('Paragraph');
  });
});
