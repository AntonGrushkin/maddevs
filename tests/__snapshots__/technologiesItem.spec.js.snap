// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechnologiesItem component renders correctly 1`] = `
<div
  class="technologies-item"
>
  <h4
    class="technologies-item__title title-md"
  >
    Title
  </h4>
   
  <p
    class="technologies-item__paragraph paragraph"
  >
    Paragraph
  </p>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <!---->
</div>
`;
